VSS.md

```diff
@@ -0,0 +1,518 @@
+
+**Replication CUJ** \- As a user, I want to rely on replication to provide high-availability and ensure that my data is protected by storing it across multiple nodes with low replication latency
+
+- Requirement: Support fast bootstrap  
+- Requirement: Support index full sync on replication lag.
+
+**Monitoring CUJ** \- As a user, I need to be able to monitor indexes and re-indexing with simple observability metrics
+
+- Requirement: Ability to monitor index base statistics  
+- Requirement: Ability to monitor cross indexes, aggregated, view statistics.
+
+**Backfill CUJ** \- As a user, I need to be able to create an index after the keyspace contains vector entries
+
+- Requirement: Index creation triggers a backfill process which indexes the existing keyspace entries.
+
+**Hybrid Query CUJ** \- As a user, I want to be able to perform hybrid queries to incorporate filtering criteria in addition to best-match vector search
+
+- Requirement: Support for filtering numeric and tag-based filters.  
+- Requirement: Support for boolean combinations of operators in combination with filtering (AND, OR, NOT)
+
+**Result Set Management CUJ** \- As a user I want to dictate which fields are returned in a search
+
+- Requirement: Support query controls to set the desired returned fields.
+
+**ACLs CUJ** \- As a user of Access Control Lists (ACLs), I want to manage user-based permissions, specific command permissions, etc to manage vector search access
+
+- Requirement: Existing ACls functionality should extend to vector search commands
+
+**Persistence CUJ** \- As a user I want to be able to leverage RDB and/or AOF for persisting and loading indexes
+
+- Requirement: Index metadata should be persisted leveraging RDB  
+- Requirement: Index types that take significant time to rebuild, such as vector indexes, must support persistence of the index payload.
+
+**Tuning CUJ** \- As a user who is also managing the infrastructure, I want to tune my configurations, like number of threads, to optimize my search performance
+
+- Requirement: Configs like number of threads should be exposed and tunable to users
+
+## Current Specification
+
+The search module allows a user to create any number of indexes. Each index consists of any number of typed fields and covers a specified subset of the keys within a Valkey database.
+
+The creation of an index initiates a backfill process in the background that will scan the keyspace and insert all covered keys into the newly created index. Mutations of a key are automatically reflected in the index. This process is automatic after the creation of the index. Mutation of keys is explicitly allowed during the backfill process. The current state of a backfill may be interrogated by a client using the FT.INFO command.
+
+A query operation is a two step process. In the first step, a query is executed that generates a set of keys and distances. In the second step, that list of keys and distances is used to generate a response.
```

Comment:
This section is about how the data is represented and stored. I miss information about how all of this is stored in RDB files, AOF files, replicated to replicas and stored in slot migration.

What are "keys and distances"?

* Are they temporary data just only produce one response and then discarded?
* Are they as actual keys in the database? If not, it should be clarified, because when the word "keys" is used, it can be confusing what it is.
* Are these keys and distances stored in RDB, AOF and replicated?
* Are they sharded or local to one node?